airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/SshTunnel.java

```diff
@@ -364,7 +364,7 @@ ClientSession openTunnel(final SshClient client) {
           remoteServiceHost, remoteServicePort, address.toInetSocketAddress()));
       return session;
     } catch (final IOException | GeneralSecurityException e) {
-      throw new RuntimeException(e);
+      throw new ConnectionErrorException(e.getMessage());
```

Comment:
I'm wondering if `ConnectionErrorException` should be extended/mutated because originally this was to handle `SQLSTATE` error codes from the database but now it's incorporating handling SSH tunnel issues which is different than the original intent

The other reason is because only surfacing `e.getMessage()` may be losing valuable information for debugging because when you get to the `catch` block of `SshWrapped<Source/Destination>` the method is already calling `e.getMessage()` so seems redundant

Based off these changes, I'm not entirely sure if we should continue to overload `ConnectionErrorException` or if this Exception should be refactored to better encapsulate the classes of configuration issues

Reply:
I think long term we may want to refactor ``ConnectionErrorException`` to better encapsulate classes of configuration issues - but that would require a detailed audit of ALL of the connection error exceptions and how they are handled. One thing we can do for now is to catch the RuntimeExceptions so that we're not creating crashing/creating alerts and providing a detailed error for the user to check their SSH connection settings

Reply:
That works for me, do you mind creating a backlog ticket that references this discussion? I should be able to approve on mobile but internet is down for me so limited bandwidth today until that gets resolved